web/client/components/misc/SecureImage.jsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright 2024, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+
+import { getAuthorizationBasic, getAuthKeyParameter, getAuthenticationMethod, getToken } from '../../utils/SecurityUtils';
+import { updateUrlParams } from '../../utils/URLUtils';
+
+
+const SecureImage = ({
+    alt,
+    src,
+    ...props
+}) => {
+    const [imageSrc, setImageSrc] = useState('');
+
+    // Function to validate the image once it loads
+    const validateImg = (imgElement) => {
+        // Implement your validation logic here
+        // For example, check image dimensions, aspect ratio, etc.
+        if (imgElement.naturalWidth === 0 || imgElement.naturalHeight === 0) {
+            console.error('Image validation failed: Image is not valid.');
+        }
+    };
+
+    useEffect(() => {
+        const authMethod = getAuthenticationMethod(src);
+        const headers = getAuthorizationBasic(props?.layer?.security?.sourceId);
+        if (props.layer) {
```

Comment:
I think the layer will be always true so probably the other `else if` have no effect with this implementation.
Shouldn't we access this first if only if `props?.layer?.security` is not undefined and `sourceId` is available?

Reply:
done